src/sentry/ingest/consumer/factory.py

```diff
@@ -159,3 +159,58 @@ def shutdown(self) -> None:
         self._pool.close()
         if self._attachments_pool:
             self._attachments_pool.close()
+
+
+class IngestTransactionsStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    """
+    Processes transactions in either celery or no-celery mode.
+    Transactions are either dispatched to `save_transaction_event` or stored directly in the
+    consumer depending on the mode.
+    """
+
+    def __init__(
+        self,
+        reprocess_only_stuck_events: bool,
+        stop_at_timestamp: int | None,
+        num_processes: int,
+        max_batch_size: int,
+        max_batch_time: int,
+        input_block_size: int | None,
+        output_block_size: int | None,
+        no_celery_mode: bool = False,
+    ):
+        self.consumer_type = ConsumerType.Transactions
+        self.reprocess_only_stuck_events = reprocess_only_stuck_events
+        self.stop_at_timestamp = stop_at_timestamp
+
+        self.multi_process = None
+        self._pool = MultiprocessingPool(num_processes)
+
+        if num_processes > 1:
+            self.multi_process = MultiProcessConfig(
+                num_processes, max_batch_size, max_batch_time, input_block_size, output_block_size
+            )
+
+        self.health_checker = HealthChecker("ingest")
+        self.no_celery_mode = no_celery_mode
```

Comment:
There is now a lot of copy paste between the IngestStrategyFactory and the IngestTransacitonStrategyFactory.
Do you plan to clean it up ?

Reply:
I think it's more likely that events and transactions will diverge as there are plans to separate more parts of the pipelines.